### PR TITLE
NovelLife Clock and option for hardcoded WIFI

### PR DESCRIPTION
**Pull request by meddle99**
Added NovelLife Clock and option for hardcoded WIFI
Also moved WIFI creds and API keys to SECRETS file -- example file included
updated readme

### CODE REVIEW

File: EleksTubeHAX/GLOBAL_DEFINES.h [hunk 1]
```diff
@@ -11,8 +11,16 @@
 #include <stdint.h>
 #include <Arduino.h>
 
-// ************ General config *********************
-//#define HARDWARE_SI_HAI_CLOCK  // comment for EleksTube original clock; uncomment only for the fake chinese copy of the clock
+
+#include "SECRETS.h"
+//// SECRETS.h must be created ---- see secretsexample_.h
```
Comment (SmittyHalibut) [inline EleksTubeHAX/GLOBAL_DEFINES.h:16]: Its a small thing, but change this to match the case of the actual file.
```suggestion
//// SECRETS.h must be created ---- see SECRETESexample.h

```

File: EleksTubeHAX/GLOBAL_DEFINES.h [hunk 2]
```diff
@@ -32,17 +40,23 @@
 #define WIFI_CONNECT_TIMEOUT_SEC  20
 #define WIFI_RETRY_CONNECTION_SEC  15
 
-#define GEOLOCATION_API_KEY " api key "  // free for 5k loopkups per month. Get yours on https://www.abstractapi.com/ (login) --> https://app.abstractapi.com/api/ip-geolocation/tester (key)
 
-// ************ MQTT config *********************
+#define HARDWARE_NO_WPS  ////comment to use WPS
```
Comment (SmittyHalibut) [inline EleksTubeHAX/GLOBAL_DEFINES.h:44]: Should this be defined inside `#define HARDWARE_NovelLife_SE_CLOCK`?  Is there a time you'd want to NOT use WPS that is not associated with the hardware used?

Also: This is `NO WPS` but everywhere it's used, it's used as `#ifndef`.  The double negative is hard to follow.  I'd prefer to see this changed to `#define HARDWARE_WPS_SUPPORT` and change the conditionals to `#ifdef` instead.  That makes more sense to me, easier to follow what's going on.

**Comment by SmittyHalibut** [comment]
I don't like the fact that the user is changing two different files: `GLOBAL_DEFINES.h` and `secretes.h`.  I think all changes we expect a user to make should be confined to a single file.

I agree that separating "bits the user changes" from "A bunch of hardware specific #defines" is a good idea, so pulling that out of `GLOBAL_DEFINES.h` is ok.  But I think ALL user changables should be pulled out.

File: README.md [hunk 3]
```diff
@@ -114,7 +115,9 @@ Make sure you configured everything in `GLOBAL_DEFINES.h`.
 * Your MQTT credentials :: Register on [SmartNest.cz](https://www.smartnest.cz/), create a Thermostat device, copy your username, API key and Thermostat Device ID.
 * Uncomment MQTT service (if in use)
 * Your Geolocation API :: Register on [Abstract API](https://www.abstractapi.com/), select Geolocation API and copy your API key.
-* Select hardware platform (original or chinese knockoff) :: comment or un-comment `#define HARDWARE_SI_HAI_CLOCK`
+* Select hardware platform (Elekstube, Novell or SI_HAI) :: un-comment appropriate hardware define 
+* Select if you prefer WPS or hardcoded credentials for WIFI
+* Create SECRETS.h file for WIFI, MQTT, and Geolocation API keys (see example file secretsexample.h)
```
Comment (SmittyHalibut) [inline README.md:120]: Match the case of the actual file.
```suggestion
* Create SECRETS.h file for WIFI, MQTT, and Geolocation API keys (see example file SECRETSexample.h)
```

**Review by SmittyHalibut — CHANGES_REQUESTED** [review]

File: EleksTubeHAX/GLOBAL_DEFINES.h (hunk 2 above)
Comment (meddle99) [reply EleksTubeHAX/GLOBAL_DEFINES.h:44]: I understand -- using ifndef seemed like a smaller change to the original code.  I can reverse it.

**Comment by meddle99** [comment]
> I moved the secrets out so i could more easily control my passwords and such, but i agree that separating the user defines makes more sense.  I wanted to do just that, but i was reticent to make so many changes to your code.  I'll work on that.  I also started working on the code for the gesture sensor.  I was able to figure out the chip that NovellLife sed (APDS-9960) and it seems interesting.

File: EleksTubeHAX/GLOBAL_DEFINES.h (hunk 2 above)
Comment (SmittyHalibut) [reply EleksTubeHAX/GLOBAL_DEFINES.h:45]: It's the same change, it's just optimizing out a double negative.

**Comment by SmittyHalibut** [comment]
> I moved the secrets out so i could more easily control my passwords and such, but i agree that separating the user defines makes more sense.  I wanted to do just that, but i was reticent to make so many changes to your code.  I'll work on that.

Improvements are always welcome. :-)

> I also started working on the code for the gesture sensor.  I was able to figure out the chip that NovellLife sed (APDS-9960) and it seems interesting.

NICE.  Like I said, improvements are always welcome.

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,2 @@
-EleksTubeHAX/global_defines.h
+EleksTubeHAX/secrets.h
```
Comment (SmittyHalibut) [inline .gitignore:1]: Just noticed this.  You're .gitignoring `secrets.h` but the file is called `SECRETES.h`.

I'm guessing you're on a computer that uses a case insensitive filesystem, aren't you?  :-)

Please keep file things case-sensitive.  Assume your users are using case sensitive filesystems.
```suggestion
EleksTubeHAX/SECRETS.h
```

File: EleksTubeHAX/GLOBAL_DEFINES.h (hunk 2 above)
Comment (meddle99) [reply EleksTubeHAX/GLOBAL_DEFINES.h:45]: I get it.  My mind isn't always linear :-)

File: .gitignore (hunk 4 above)
Comment (meddle99) [reply .gitignore]: secrets.h (and all capitalization variances :-) ) removed

**Comment by meddle99** [comment]
I think I resolved all the points

**Review by SmittyHalibut — APPROVED** [review]
Looks good to me.  I haven't TESTED this code, but I don't see anything concerning anymore.  Thanks for all the changes. :-)

**Comment by aly-fly** [comment]
> Added NovelLife Clock ...
@meddle99 Do you have any photos of the clock mainboard, preferably empty board, photo taken directly from the top? It would be great to have it, to add it in the photo section of the supported clocks.
